### PR TITLE
Add port_type argument to os_port module

### PR DESCRIPTION
**Pull request by MaxBab**
##### SUMMARY
Openstack has multiple types for the port.
Add type selection during the port creation and update.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
os_port

##### ANSIBLE VERSION
ansible 2.7.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/home/stack/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /tmp/ansible_venv/lib/python2.7/site-packages/ansible
  executable location = /tmp/ansible_venv/bin/ansible
  python version = 2.7.5 (default, May 31 2018, 09:41:32) [GCC 4.8.5 20150623 (Red Hat 4.8.5-28)]

##### ADDITIONAL INFORMATION
Depends-On: https://review.openstack.org/#/c/612162/

### CODE REVIEW

File: lib/ansible/modules/cloud/openstack/os_port.py [hunk 1]
```diff
@@ -78,6 +78,12 @@
    availability_zone:
      description:
        - Ignored. Present for backwards compatibility
+   vnic_type:
+     description:
+       - The type of the port tha tshould be created
```
Comment (mnaser) [inline lib/ansible/modules/cloud/openstack/os_port.py]:
```suggestion
       - The type of the port that should be created
```

**Review by mnaser — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/cloud/openstack/os_port.py (hunk 1 above)
Comment (MaxBab) [reply lib/ansible/modules/cloud/openstack/os_port.py]: Hi @mnaser ,
Thanks for the review.
Fixed.
In addition, changed the code to be cleaner.

**Comment by MaxBab** [comment]
Hello @mnaser ,

Could you take a look at the patch, please?

Thank you.

**Comment by mnaser** [comment]
shipit

**Comment by mnaser** [comment]
Thanks @MaxBab

**Comment by MaxBab** [comment]
Hello @emonty ,

Could you take a look at the patch, please?

Thank you.

**Comment by MaxBab** [comment]
Hello @mnaser ,

Is there any other person I could ask for the review?

Thank you.

**Comment by MaxBab** [comment]
Hello @odyssey4me ,

Could you take a look at the patch, please?

Thank you.

**Review by odyssey4me — APPROVED** [review]

**Comment by odyssey4me** [comment]
ship

**Comment by odyssey4me** [comment]
shipit

**Comment by MaxBab** [comment]
Hello @mnaser @odyssey4me ,

I rebased the patch according to the latest changes.
Could you take a look once again, please?

Thank you.

**Comment by odyssey4me** [comment]
shipit

**Comment by MaxBab** [comment]
Hello @gundalow @abadger  ,

Could you take a look at the patch, please?

Thank you.

**Comment by Yarboa** [comment]
Looks, Good to me, this fix, is required 
@gundalow @abadger , can you take closer look?

**Review by gundalow — APPROVED** [review]

**Comment by gundalow** [comment]
Thanks, merged into devel for release in Ansible 2.8.

**Pull request merged**